Automattic/wp-calypso
# Framework: add new BackButton component

In Reader, we originally created a back button component (`blocks/reader-full-post/back`) to use in full post view. That component started to be used outside Reader (see https://github.com/Automattic/wp-calypso/pull/22723) and its styles were only present in the full post stylesheet, so it seemed an excellent time to extract the back button to its own Calypso component.

<img width="82" alt="screen shot 2018-02-23 at 14 55 12" src="https://user-images.githubusercontent.com/17325/36577150-9389028a-18a9-11e8-936d-672d7857a21d.png">

This PR:
- adds the new BackButton component http://calypso.localhost:3000/devdocs/design/back-button
- removes the old ReaderFullPostBack component
- updates uses of ReaderFullPostBack to use BackButton instead

Happily, this also seems to fix #19948 (back link not appearing on mobile).

### To test

Navigate to a full post in Reader, and check that the back link appears at the top left and functions correctly.

At mobile widths, the label should disappear leaving the arrow only:

<img width="393" alt="screen shot 2018-02-23 at 15 02 10" src="https://user-images.githubusercontent.com/17325/36577295-9358ad78-18aa-11e8-8f8a-61d5949b65d8.png">
